hw3/hw3.py

```diff
@@ -0,0 +1,45 @@
+# '''
+# Создать форму для регистрации пользователей на сайте. Форма должна содержать поля "Имя", "Фамилия", "Email", "Пароль"
+# и кнопку "Зарегистрироваться". При отправке формы данные должны сохраняться в базе данных,
+# а пароль должен быть зашифрован.
+# '''
+
+from flask import Flask, render_template, redirect, url_for, request
+from flask_wtf.csrf import CSRFProtect
+from werkzeug.security import generate_password_hash
+# from werkzeug.security import check_password_hash
+
+from models import db, User
+from forms import Registration
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///users.db'
+app.secret_key = b'7435220954b80e033873947b65db8d28fd9d5d8b3f90ea952a73f60fd76c2366'
+csrf = CSRFProtect(app)
+# csrf.init_app(app)
+
+db.init_app(app)
+
+
+@app.cli.command('create')
+def create():
+	db.create_all()
+	print('CREATION COMPLETE')
+
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+	form = Registration()
+	# if form.validate_on_submit():  # если нажата кнопка (она может быть нажата, только если все поля проверены)
+	# этот вариант выдавал ошибку "flask method not allowed"
+	if request.method == 'POST' and form.validate:
```

Comment:
Ноя  бы все-таки использовал метод `validate_on_submit`
```suggestion
	if form.validate_on_submit():  # если нажата кнопка (она может быть нажата, только если все поля проверены)
```

Reply:
У меня как раз form.validate_on_submit() и вызывал ошибку method not allowed. Попробую применить все ваши исправления, может заработает